symengine/tests/basic/test_cse.cpp

```diff
@@ -302,6 +306,30 @@ TEST_CASE("CSE: simple", "[cse]")
         REQUIRE(unified_eq(substs, {{x0, add(x, y)}, {x1, add(x0, z)}}));
         REQUIRE(unified_eq(reduced, {x0, add(i2, x0), x1, add(i3, x1)}));
     }
+    {
+        auto pw1 = piecewise(
+            {{pow(add(x, y), i2), Gt(x, y)}, {sqrt(add(x, y)), boolTrue}});
+
+        vec_pair substs;
+        vec_basic reduced;
+        cse(substs, reduced, {pw1});
+        REQUIRE(unified_eq(substs, {{x0, add(x, y)}}));
+        REQUIRE(unified_eq(reduced, {piecewise({{pow(x0, i2), Gt(x, y)},
+                                                {sqrt(x0), boolTrue}})}));
+    }
+    {
+        auto pw2 = piecewise({{pow(x, i2), Gt(add(x, y), i2)},
+                              {sqrt(y), Gt(add(x, y), i3)},
```

Comment:
Can you swap the two conditions? They don't make much sense.

Reply:
Sorry about that, fixed in 3570b305a835f3d57d71d98c6d6ed3ee87863eb0

Reply:
Looks like you only changed part of the test